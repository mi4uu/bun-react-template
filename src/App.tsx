import { Counter } from './Counter/Counter'
import style from './App.module.css'
export const App = () => {
  return (
    <div className={style.App} role='main'>
      <article className={style.AppArticle}>
        <img src={ '/bunlogo.svg' } className={style.AppLogo} alt='logo' />
        <div className={ style.br0 }/>
        <h3>
          Welcome to Bun!
        </h3>
        <div className={ style.br1 }>
        </div>
        <a className={style.AppLink} href='https://bun.sh/docs' target='_blank' rel='noopener noreferrer'>
          Read the docs â†’
        </a>
        <Counter />
      </article>
    </div>
  )
}
