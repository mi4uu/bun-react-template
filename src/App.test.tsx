import { afterEach, beforeEach, expect, test } from 'bun:test'
import { createRoot } from 'react-dom/client'

import { act } from 'react-dom/test-utils'

import assert from 'assert'
import { App } from './App'
import React from 'react'

let container : HTMLDivElement | null = null
beforeEach( () => {
  // setup a DOM element as a render target
  container = document.createElement( 'div' )
  document.body.appendChild( container )
} )

afterEach( () => {
  // cleanup on exiting
  if ( container ) {
    document.body.removeChild( container )
    container = null
  }
} )

test('App', () => {
  act( () => {
    createRoot( container! ).render( <App /> )
  } )

 expect( container?.children[0].innerHTML ).toMatchSnapshot()
 assert( container )

  const article = container!.querySelector( 'article' )
  expect( article?.children.length ).toBe( 6 )
})
