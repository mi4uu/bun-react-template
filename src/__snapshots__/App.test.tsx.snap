// Bun Snapshot v1, https://goo.gl/fbAQLP

exports[`App 1`] = `"<article><img alt="logo" src="/bunlogo.svg"><div></div><h3>Welcome to Bun!</h3><div></div><a href="https://bun.sh/docs" target="_blank" rel="noopener noreferrer">Read the docs â†’</a><div data-testid="counter" class=""><a>-</a><b>0</b><a>+</a></div></article>"`;
